go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -407,12 +407,7 @@ func getEvalEngingeExpr(ctx *plancontext.PlanningContext, pe *operators.ProjExpr
 
 // useSimpleProjection uses nothing at all if the output is already correct,
 // or SimpleProjection when we have to reorder or truncate the columns
-func useSimpleProjection(ctx *plancontext.PlanningContext, op *operators.Projection, cols []int, colNames []string, src logicalPlan) (logicalPlan, error) {
-	columns := op.Source.GetColumns(ctx)
-	if len(columns) == len(cols) && elementsMatchIndices(cols) {
-		// the columns are already in the right order. we don't need anything at all here
-		return src, nil
-	}
+func useSimpleProjection(cols []int, colNames []string, src logicalPlan) (logicalPlan, error) {
 	return &simpleProjection{
```

Comment:
should we not check for output column names so that we can still simplify for cases where there is no column alias.

Reply:
we shouldn't do that here in the `operator_transformers.go`. If we failed to do it while planning on the operators, we're not going to succeed here.

Reply:
We are doing that check already in `compact` stage of the operator tree.